web/src/engine/websites/Mikoroku.ts

```diff
@@ -0,0 +1,56 @@
+import { Tags } from '../Tags';
+import icon from './Mikoroku.webp';
+import { DecoratableMangaScraper, type MangaPlugin, Manga, Chapter } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchWindowScript } from '../platform/FetchProvider';
+
+const pagesScript = `
+    new Promise(resolve => { resolve([...document.querySelectorAll('article#reader img')].map(img => img.src)); });
+`;
+
+type RSS = {
+    feed: {
+        entry: {
+            title: {
+                $t : string
+            },
+            link: {
+                rel: string
+                title: string,
+                href: string
+            }[]
+        }[]
+    }
+}
+
+@Common.MangaCSS(/^{origin}\/\d+\/\d+\/[^/]+\.html$/, 'header h1[itemprop="name"]')
+@Common.PagesSinglePageJS(pagesScript, 500)
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mikoroku', 'Mikoroku', 'https://www.mikoroku.web.id', Tags.Media.Manga, Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Indonesian, Tags.Source.Scanlator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const { feed: { entry } } = await FetchJSON<RSS>(new Request(new URL('/feeds/posts/default/-/Series?orderby=published&alt=json&max-results=999', this.URI)));
+        return entry.map(manga => {
+            const goodLink = manga.link.find(link => link.rel === 'alternate');
+            return new Manga(this, provider, new URL(goodLink.href).pathname, goodLink.title.trim());
```

Comment:
Would it be possible to determine the real manga id instead of just a link which requires the injection of an additional script in `FetchChapters` to get the real manga id?

Reply:
From my experience with NextScan , manga id for this request is a real mess. You think its the slug, but sometimes its  the title (o the same website) , and dont let me talk about the non-consistant CaSeS problems. NextScans gave me headaches.

I did that because i want it to be reliable.

Reply:
![image](https://github.com/user-attachments/assets/48974c36-f78d-4ef2-85bb-8e21928a3dcd)

That is the title : `Er0t1c4l Wizard & Twelve Brides`.
slug is `erotical-wizard-twelve-brides`.

You may say : both of them works? Well, i'd found (on NextScans) thats kinda random

Reply:
Wait for this website there is an alernative